### ISSUE TITLE
intermittent scala-2.11.x-integrate-ide failures

### ISSUE THREAD

**Issue by SethTisue** [scala/scala-dev#76]
```
[INFO] org.scala-ide.sdt.core.tests ...................... FAILURE [15:03.618s]
```

examples:
- https://scala-ci.typesafe.com/job/scala-2.11.x-integrate-ide/2039/console
- https://scala-ci.typesafe.com/job/scala-2.11.x-integrate-ide/2038/console
- https://scala-ci.typesafe.com/job/scala-2.11.x-integrate-ide/2037/console

@sschaef, @dragos, any idea what this is about? there are so many megabytes of log output I couldn't even find where the actual failure is.

**Comment by kiritsuku** [comment]
\cc @wpopielarski

These are all flaky tests. We should discuss what we can do in future to avoid having them and how we can in general make our testing infrastructure more stable.

**Comment by kiritsuku** [comment]
Btw, the failed tests can be seen in a summary at the end of the log. No need to read or even understand all of it. ;)

**Comment by SethTisue** [comment]
> the failed tests can be seen in a summary at the end of the log

maybe it's staring me in the face, but I'm not seeing it. can you show me the lines you mean?

**Comment by kiritsuku** [comment]
Beside from a lot of other noise that is shown by maven, this is the essence:

```
Tests run: 1049, Failures: 0, Errors: 1, Skipped: 46, Time elapsed: 761.622 sec <<< FAILURE!

Results :

Tests in error: 
  multiple_FJScheduler(org.scalaide.core.interpreter.EclipseReplTest): Futures timed out after [200 milliseconds]

Tests run: 1049, Failures: 0, Errors: 1, Skipped: 46

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] org.scala-ide.sdt.build ........................... SUCCESS [4.566s]
[INFO] org.scala-ide.sdt.aspects ......................... SUCCESS [26.384s]
[INFO] org.scala-ide.sdt.core ............................ SUCCESS [3:21.765s]
[INFO] org.scala-ide.sdt.core.tests ...................... FAILURE [15:03.618s]
[INFO] org.scala-ide.sdt.spy ............................. SKIPPED
[INFO] org.scala-ide.sdt.debug ........................... SKIPPED
[INFO] org.scala-ide.sdt.debug.tests ..................... SKIPPED
[INFO] org.scala-ide.sdt.debug.expression ................ SKIPPED
[INFO] org.scala-ide.sdt.debug.expression.tests .......... SKIPPED
[INFO] org.scala-ide.sdt.weaving.feature ................. SKIPPED
[INFO] org.scala-ide.sdt.feature ......................... SKIPPED
[INFO] org.scala-ide.sdt.dev.feature ..................... SKIPPED
[INFO] org.scala-ide.sdt.source.feature .................. SKIPPED
[INFO] org.scala-ide.sdt.update-site ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 23:01.399s
[INFO] Finished at: Mon Jan 25 19:36:07 UTC 2016
[INFO] Final Memory: 151M/1007M
```

Not exactly the end of the log but it is not too far away from it.

**Comment by SethTisue** [comment]
ah, well yeah, I quoted that much in my first comment above. wondering what failed more specifically than just "core.tests"

**Comment by dragos** [comment]
Futures timed out after [200 milliseconds]

Is this a very slow machine? Should we simply have even larger timeouts?

On Tue, Jan 26, 2016 at 2:55 AM, Seth Tisue notifications@github.com
wrote:

> ah, well yeah, I quoted that much in one of my comments above. wondering
> what failed more specifically.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/scala/scala-dev/issues/76#issuecomment-174770594.

## 

« Je déteste la montagne, ça cache le paysage »
Alphonse Allais

**Comment by SethTisue** [comment]
> Is this a very slow machine

Well, our Jenkins workers are sometimes pretty heavily loaded. For example, Jenkins will run the Scala community build in parallel with validating a PR on the same worker.

Running tests under varying loads seems like a feature, to me, on the balance. So yeah, maybe longer timeouts.

**Comment by kiritsuku** [comment]
no longer relevant
